modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/custom/form/components/PartialResults.tsx

```diff
@@ -29,7 +29,21 @@ const PartialResults: React.FC<IProps> = ({
 	onShow,
 	reportDataURL,
 }) => {
-	const {resource} = useResource({fetch, link: reportDataURL});
+	const [resourceState, setResourceState] = useState(() => ({
+		error: false,
+		loading: true,
+	}));
+	const {error, loading} = resourceState;
+	const {resource} = useResource({
+		fetch,
+		link: reportDataURL,
+		onNetworkStatusChange: (status) =>
+			setResourceState({
+				error: status === 5,
+				loading: status < 4,
+			}),
+	});
```

Comment:
But it think would be simpler if we handle the `error` and the `loading` as a  single state:
```suggestion
	const [resourceState, setResourceState] = useState(() => ('loading'));
	const {resource} = useResource({
		fetch,
		link: reportDataURL,
		onNetworkStatusChange: (status) => {
			let resourceState = 'idle';			
			if (status < 4) {
				resourceState = 'loading'
			}
			if (status === 5) {
				resourceState = 'error'
			}
			setResourceState(resourceState);
		},
	});
```